lint

```diff
@@ -172,6 +174,34 @@ lint_files() {
     exit $lint_result
 }
 
+matches_any() {
+    local filename="$1"
+    local patterns="$2"
+    IFS=$'\n' # Make newlines the only separator.
+    set -f    # Disable globbing so $patterns can be treated as is.
+    for pattern in $patterns; do
+        # Use the [[ operator without quotes on $pattern 
+        # in order to "glob" the provided filename:
+        [[ "$filename" == $pattern ]] && return 0
+    done
+    return 1
+}
+
+filter_out() {
+    local patterns="$1"
+    if [ -n "$patterns" ] && [ -r "$patterns" ]; then
+        local patterns=$(sed '/^#.*$/d ; /^\s*$/d' $patterns) # Remove blank lines and comments before we start iterating.
+        [ -n "$DEBUG" ] && echo >&2 "> Filters:" && echo >&2 "$patterns"
+        local filtered_out=()
+        while read -r filename; do
+            matches_any "$filename" "$patterns" && filtered_out+=("$filename") || echo "$filename"
```

Comment:
I'd prefer a long-form `if...else...fi` for this condition, since it's not immediately obvious that that's what this is doing